runner/execer/os/execer.go

```diff
@@ -157,11 +161,25 @@ func (e *execer) monitorMem(p *process, memCh chan scootexecer.ProcessStatus) {
 			if _, err := e.pw.GetProcs(); err != nil {
 				log.Error(err)
 			}
-			mem, _ := e.pw.MemUsage(pid)
+			var mem scootexecer.Memory
+			// if not nil, use the provided function to get memory utilization,
+			// otherwise get the memory usage of the current process and its subprocesses
+			if e.getMemUtilization != nil {
+				var memory int64
+				memory, err = e.getMemUtilization()
+				mem = scootexecer.Memory(memory)
+			} else {
+				mem, err = e.pw.MemUsage(pid)
```

Comment:
Just one style comment: is there a way we can initialize such that we assign getMemUtilization() to some wrapper that calls e.pw.MemUsage(pid) in this case, and we don't need the if/else branch here?

Reply:
Moved to NewBoundedExecer() func.